alembic/ddl/postgresql.py

```diff
@@ -218,7 +218,8 @@ def autogen_column_reflect(self, inspector, table, column_info):
                         "join pg_class t on t.oid=d.refobjid "
                         "join pg_attribute a on a.attrelid=t.oid and "
                         "a.attnum=d.refobjsubid "
-                        "where c.relkind='S' and c.relname=:seqname"
+                        "where c.relkind='S' and "
+                        "c.oid=cast(:seqname as regclass)"
```

Comment:
**Michael Bayer** (zzzeek) wrote:

how on earth does this work?   I have no idea how this works, PG does indeed return true for this

    (Pdb) sql = "select c.relname, c.oid, cast('mytable_id_seq' as regclass), c.oid=cast('mytable_id_seq' as regclass), a.attname from pg_class as c join pg_depend d on d.objid=c.oid and d.classid='pg_class'::regclass and d.refclassid='pg_class'::regclass join pg_class t on t.oid=d.refobjid join pg_attribute a on a.attrelid=t.oid and a.attnum=d.refobjsubid where c.relkind='S' and c.oid=cast('mytable_id_seq' as regclass)"
    (Pdb) conn.exec_driver_sql(sql).all()
    [('mytable_id_seq', 19122, 'mytable_id_seq', True, 'id')]


c.oid is 19122 and cast('mytable_id_seq' as regclass) is "mytable_id_seq", yet they compare as True????   WTF ????

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5333

Reply:
isnt there a more explicit `to_oid` kind of function we can use ?  it's totally acting like magic here

Reply:
**Michael Bayer** (zzzeek) wrote:

documented here: https://www.postgresql.org/docs/current/datatype-oid.html

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5333

Reply:
> **Michael Bayer** (zzzeek) wrote:
> 
> documented here: https://www.postgresql.org/docs/current/datatype-oid.html
> 
> View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5333

oh right, I made improve that page, since IIRC I opened a bug report since it was not documented anywhere

Reply:
You got them to document that ?  Awesome

Reply:
They improved them following this issue https://www.postgresql.org/message-id/16991-bcaeaafa17e0a723%40postgresql.org